linode/manager
# refactor: [M3-8877] - Refactor VPC Create to use `react-hook-form`

## Description 📝
- Use react-hook-form for VPC Create

## Changes  🔄
- Switched VPC create forms to react-hook-form and cleaned up accordingly
- update VPC create unit tests

## Target release date 🗓️
n/a

## How to test 🧪
- confirm behavior matches prod
  - error validation + scrolling into error
  - subnet recommendations + # of IPs 
- confirm tests pass:
```
yarn test VPCCreate
```
```
yarn cy:run -s "cypress/e2e/core/vpc/vpc-create.spec.ts"
```
```
yarn cy:run -s "cypress/e2e/core/linodes/create-linode-with-vpc.spec.ts"
```


<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support

<br/>


## As an Author, before moving this PR from Draft to Open, I confirmed ✅


</details>
